### PR TITLE
Add switch to function addLayerToMesh to copy the MaterialID.

### PR DESCRIPTION
**Pull request by TomFischer**
In some applications it is practical to adopt the material ID of the underlying layer.

### CODE REVIEW

File: Applications/DataExplorer/DataView/MeshView.cpp [hunk 1]
```diff
@@ -244,9 +244,11 @@ void MeshView::openAddLayerDialog()
         return;
     }
 
-    double const thickness (dlg.getThickness());
-    std::unique_ptr<MeshLib::Mesh> result(MeshLib::addLayerToMesh(
-        *mesh, thickness, dlg.getName(), dlg.isTopLayer()));
+    bool const take_over_material_ids = false;
```
Comment (endJunction) [inline Applications/DataExplorer/DataView/MeshView.cpp:247]: Take over the world, that's right. Here pass material ids, or copy material ids would be better, before we take over the world. It's a funny evening today ;)

File: MeshLib/MeshEditing/AddLayerToMesh.cpp [hunk 2]
```diff
@@ -192,12 +194,25 @@ MeshLib::Mesh* addLayerToMesh(MeshLib::Mesh const& mesh, double thickness,
...
+        sfc_mesh->getProperties().existsPropertyVector<int>("MaterialIDs"))
+    {
+        auto const& sfc_material_ids =
+            *sfc_mesh->getProperties().getPropertyVector<int>("MaterialIDs");
+        std::copy(sfc_material_ids.cbegin(), sfc_material_ids.cend(),
+                  std::back_inserter(*new_materials));
+    }
+    else
+    {
+        int new_layer_id(
```
Comment (endJunction) [inline MeshLib/MeshEditing/AddLayerToMesh.cpp:211]:
```suggestion
        int const new_layer_id(
```

File: Applications/DataExplorer/DataView/MeshView.cpp (hunk 1 above)
Comment (TomFischer) [reply Applications/DataExplorer/DataView/MeshView.cpp]: Renamed as suggested.

File: MeshLib/MeshEditing/AddLayerToMesh.cpp (hunk 2 above)
Comment (TomFischer) [reply MeshLib/MeshEditing/AddLayerToMesh.cpp]: Const added.

**Comment by endJunction** [comment]
If no other reviews, will be merged tonight.

**Pull request merged**